tools/pubsys/src/aws/publish_ami/mod.rs

```diff
@@ -200,18 +203,32 @@ pub(crate) async fn run(args: &Args, publish_args: &PublishArgs) -> Result<()> {
     .await?;
 
     info!("Updating AMI permissions - {}", description);
-    let ami_ids = amis
-        .into_iter()
-        .map(|(region, image)| (region, image.id))
-        .collect();
     modify_regional_images(
         &publish_args.modify_opts,
         &operation,
-        &ami_ids,
+        &mut amis,
         &ec2_clients,
     )
     .await?;
 
+    write_amis(&publish_args.ami_input, &amis)?;
+
+    Ok(())
+}
+
+fn write_amis(ami_output: &PathBuf, amis: &HashMap<Region, Image>) -> Result<()> {
+    let file = File::create(ami_output).context(error::FileSnafu {
+        op: "write amis to file",
+        path: ami_output,
```

Comment:
This function should probably also be [used in `pubsys ami` command](https://github.com/bottlerocket-os/bottlerocket/pull/2987/commits/2254cb6fadc000b2915d634042c88c9e5f823101#diff-ce04b11594176c7886cc95d134e3d1100b817c0986274d5adde9889b4821c225R81-R89), right?